.github/workflows/build_and_test.yml

```diff
@@ -153,91 +148,79 @@ jobs:
     - name: Build and run tket proptests
       if: needs.check_changes.outputs.tket_changed == 'true' || needs.check_changes.outputs.tket_proptests_changed == 'true'
       run: conan create --profile=tket recipes/tket-proptests --build=missing --build=tket-proptests
+    - name: Upload tket conan package
+      uses: actions/upload-artifact@v3
+      with:
+        name: tket_conan_${{ matrix.os }}
+        path: ~/.conan/data/tket
+    - name: Upload package
+      if: github.event_name == 'push' && github.ref == 'refs/heads/develop' && needs.check_changes.outputs.tket_changed == 'true'
+      run: |
+        conan user -p ${{ secrets.JFROG_ARTIFACTORY_TOKEN_2 }} -r tket-libs ${{ secrets.JFROG_ARTIFACTORY_USER_2 }}
+        conan upload tket/${{ needs.check_changes.outputs.tket_ver }}@tket/stable --all -r=tket-libs
+
+  build_test:
+    name: Build and test
+    needs: build_test_tket
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04', 'macos-11', 'windows-2022']
+        pyver: ['3.8', '3.9', '3.10']
+    runs-on: ${{ matrix.os }}
+    env:
+      CONAN_REVISIONS_ENABLED: 1
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install conan
+      uses: turtlebrowser/get-conan@v1.1
+    - name: Set up conan
+      shell: bash
+      run: ./.github/workflows/conan-setup
+    - name: Download tket conan package
+      uses: actions/download-artifact@v3
+      with:
+        name: tket_conan_${{ matrix.os }}
+        path: ~/.conan/data/tket
     - name: Install pybind11
       run: conan create --profile=tket recipes/pybind11
-    - name: Set up Python 3.8
-      if: github.event_name == 'push'
+    - name: Set up Python
       uses: actions/setup-python@v3
       with:
-        python-version: '3.8'
-    - name: Build pytket (3.8)
-      if: github.event_name == 'push'
+        python-version: ${{ matrix.pyver }}
+    - name: Build pytket
       run: |
         cd pytket
         pip install -e . -v
-    - name: Test pytket (3.8) with coverage
-      if: matrix.os == 'ubuntu-22.04' && github.event_name == 'push'
+    - name: Test pytket with coverage
+      if: matrix.os == 'ubuntu-22.04' && matrix.pyver == '3.8'
       run: |
         cd pytket/tests
         pip install -r requirements.txt
         pytest --ignore=simulator/ --doctest-modules --cov=../pytket --cov-branch --cov-report=html --cov-report=xml:htmlcov/cov.xml
-    - name: Test pytket (3.8)
-      if: matrix.os != 'ubuntu-22.04' && github.event_name == 'push'
+    - name: Test pytket
+      if: matrix.os != 'ubuntu-22.04' || matrix.pyver != '3.8'
```

Comment:
Why are we doing the exclusion?

Reply:
is this the case where the coverage check is done?

Reply:
Yes, the tests are already run in that case using the coverage flags, so no need to run them again.